source/NRF52Serial.cpp

```diff
@@ -296,3 +296,55 @@ void NRF52Serial::updateRxBufferAfterRXSTARTED()
 {
     nrf_uarte_rx_buffer_set(p_uarte_, dmaBuffer, CONFIG_SERIAL_DMA_BUFFER_SIZE);
 }
+
+/**
+ * Puts the component in (or out of) sleep (low power) mode.
+ */
+int NRF52Serial::setSleep(bool doSleep)
+{
+    IRQn_Type IRQn = get_alloc_peri_irqn(p_uarte_);
+
+    if (doSleep)
+    {
+        if ( !(status & CODAL_SERIAL_STATUS_DEEPSLEEP))
+        {
+            if(txInUse() || rxInUse())
+                return DEVICE_SERIAL_IN_USE;
+
+            lockTx();
+            lockRx();
+
+            if(txBufferedSize() > 0)
+                disableInterrupt(TxInterrupt);
```

Comment:
Is this right @martinwork? Looks like we're intentionally disabled TX interrupts here when there's still data to send??

Reply:
Good point @finneyj! disableInterrupt(TxInterrupt) does nothing though.  See configurePins.

I'll swap it round so it makes more sense. Perhaps we do actually want to disable the interrupt here?

Reply:
Hmm... If it does nothing, do we need this bit of code at all?

Reply:
True!